### ISSUE TITLE
Unable to restart sshd on el7

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#947]
Originally reported of Google Code with ID 891.

```
What software version are you running?
gnt-cluster (ganeti v2.11.3) 2.11.3
Software version: 2.11.3
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.3

<b>What distribution are you using?</b>
CenOS 7 / RHEL7

<b>What steps will reproduce the problem?</b>
1. gnt-node add NODENAME

What do you see instead?
/usr/lib64/ganeti/daemon-util: line 371: /etc/rc.d/init.d/sshd: No such file or directory
2014-07-19 15:05:35,039: Unhandled Ganeti error: Could not reload SSH keys, command '/usr/lib64/ganeti/daemon-util reload-ssh-keys' had exitcode 1 and error 
Failure: command execution error:
Command 'ssh -oEscapeChar=none -oHashKnownHosts=no -oGlobalKnownHostsFile=/var/lib/ganeti/known_hosts -oUserKnownHostsFile=/dev/null -oCheckHostIp=no -oPort=22 -oStrictHostKeyChecking=ask -4 root@NODENAME '/bin/sh -c '\''test -d /usr/lib64/ganeti/2.11 && ln -s -f -T /usr/lib64/ganeti/2.11 /etc/ganeti/lib && ln -s -f -T /usr/share/ganeti/2.11 /etc/ganeti/share && /usr/lib64/ganeti/prepare-node-join'\'''' failed: exited with exit code 1


<b>Please provide any additional information below.</b>
el7 changed Daemon-Management to systemd-based - Patch from Fedora should work https://github.com/jfut/ganeti-rpm/blob/master/rpmbuild/ganeti/SOURCES/ganeti-2.8.0-fedora.patch
```
Originally added on 2014-07-19 13:15:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I wonder whether the --with-ssh-initscript configure option can
be used to solve your problem in a clean way. Also note that Ganeti
from 2.12 onwards (yes, only about to reach beta1) has support for use
with systemd; Apollon should know most about the details of systemd
support.
```
Originally added on 2014-07-21 12:29:17 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unfortunately you cannot use --with-ssh-initscript directly with systemd, because the arguments in systemd are in reverse order ("systemctl restart sshd.service"). On Debian it will work because we have transparent wrappers around the initscripts and/or invoke-rc.d.

What ganeti could do to resolve this, would be to deprecate --with-ssh-initscript and replace it with "--with-ssh-restart-command" and accept the full command restarting the ssh service.
```
Originally added on 2014-07-21 13:04:08 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-07-21 13:16:18 +0000 UTC.

Added Labels: Type-Enhancement Priority-Low
Changed State: Accepted
Added to Milestone: Release2.15

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-06-10 13:33:26 +0000 UTC.

Removed Labels: Priority-Low
Changed State: PatchesWelcome
Added to Milestone: Unplanned
Removed from Milestone: Release2.15

**Comment by ge-fa** [comment]
Is this still valid as of today? I only very rarely deal with CentOS..

**Comment by apoikos** [comment]
@ge-fa: as far as I can tell, yes, it's still valid. I think the most straightforward change would be to add a `--with-ssh-service` configure parameter and then use `service <X> restart` which works well across distributions and init systems.

**Comment by apoikos** [comment]
Then again, I would be happy if Ganeti didn't have to touch the SSH config to work.

**Comment by mbakke** [comment]
I believe this was fixed by @rbott in 345b29addde6437866395ea078854c50185f8742.

**Comment by saschalucas** [comment]
@mbakke indeed, I confirm, that this should be fixed by @rbott in 345b29a. Thanks.
